### PR TITLE
Negative Y Values Support and Example story for VBC

### PR DESCRIPTION
**Pull request by Anush2303**
image

## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by AtishayMsft** [comment]
For this PR target to master

**Comment by fabricteam** [comment]
## 🕵 FluentUIV0 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v0"/>

File: apps/public-docsite-resources/config/api-docs.js [hunk 1]
```diff
@@ -108,6 +108,7 @@ module.exports = {
       'MultiStackedBarChart',
```
Comment (fabricteam) [inline apps/public-docsite-resources/config/api-docs.js:108]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33369&commitId=e6df356a163b303dd38ee31308284479b212d844&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|8| Changed|

</details>


  <details><summary><b>Drawer</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.Full Overlay High Contrast.chromium.png|2213| Changed|
|Drawer.overlay drawer full.chromium.png|1150| Changed|
|Drawer.Full Overlay RTL.chromium.png|1168| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/charts/react-charting/src/utilities/utilities.ts [hunk 2]
```diff
@@ -437,6 +447,7 @@ export const createStringYAxisForHorizontalBarChartWithAxis = (
   dataPoints: string[],
   isRtl: boolean,
   barWidth: number,
+  supportNegativeValuesForYAxis: boolean = false,
```
Comment (srmukher) [inline packages/charts/react-charting/src/utilities/utilities.ts:450]: is this change required?

File: packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+   * The schema representing the chart
+   */
+  chartSchema: any;
+}
+
+/**
+ * DeclarativeChart component.
+ * {@docCategory DeclarativeChart}
+ */
+export const DeclarativeChart: React.FunctionComponent<DeclarativeChartProps> = React.forwardRef<
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx:39]: This PR should have only negative value changes. And not declarative chart changes.

Comment (Anush2303) [reply packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx:39]: Okay. Closing this PR.

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 4]
```diff
@@ -700,18 +707,23 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
         legendColor: this.state.color,
         shouldHighlight,
       });
-      const barHeight: number = Math.max(yBarScale(point.y), 0);
-      let adjustedBarHeight = 0;
-
-      if (barHeight <= 0) {
+      const barHeight: number = this.props.supportNegativeYValues
+        ? yBarScale(point.y) - yBarScale(0)
+        : yBarScale(point.y);
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:712]: why remove math.max.

File: packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx (hunk 3 above)
Comment (Anush2303) [reply packages/charts/react-charting/src/components/DeclarativeChart/DeclarativeChart.tsx:39]: New PR: [33377](https://github.com/microsoft/fluentui/pull/33377)

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 5]
```diff
@@ -700,18 +707,23 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
...
+      // Calculate threshold for minimum visible bar height
+      const minBarHeight = Math.ceil(yBarScale(this._yMax - this._yMin) / 100.0);
+      let adjustedBarHeight = barHeight;
+
+      if (Math.abs(barHeight) === 0 || (barHeight < 0 && !this.props.supportNegativeYValues)) {
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:717]: barHeight <= 0 should be a sufficient condition.

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 4 above)
Comment (Anush2303) [reply packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:712]: Math.max function was necessary before because the chart didn't handle negative values. When negative values were encountered, it would force the bar height to 0, ensuring that no bars went below the baseline.
But now new logic explicitly handles the negative values by adjusting the scale calculation.

File: packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 5 above)
Comment (Anush2303) [reply packages/charts/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:717]: using barHeight<=0 will also lead to negative height bars to not render. But now we don't want that. We will now not render anything for barHeight =0 or if (negative value prop is false and bar height is negative).

File: packages/charts/react-charting/src/utilities/utilities.ts [hunk 6]
```diff
@@ -347,7 +347,13 @@ export function prepareDatapoints(
     : (maxVal - minVal) / splitInto >= 1
     ? Math.ceil((maxVal - minVal) / splitInto)
     : (maxVal - minVal) / splitInto;
-  const dataPointsArray: number[] = [minVal, minVal + val];
+  const dataPointsArray: number[] = [minVal < 0 && maxVal >= 0 ? 0 : minVal];
+  if (minVal < 0 && maxVal >= 0) {
```
Comment (AtishayMsft) [inline packages/charts/react-charting/src/utilities/utilities.ts:351]: what happens if both minVal and maxVal < 0

Comment (Anush2303) [reply packages/charts/react-charting/src/utilities/utilities.ts:351]: For (minVal<0 and maxVal<0) and (minVal>0 and maxVal>0)
the default while loop will run

Comment (Anush2303) [reply packages/charts/react-charting/src/utilities/utilities.ts:351]: Line 357
